src/com/google/enterprise/adaptor/ContentTransformFactory.java

```diff
@@ -0,0 +1,75 @@
+package com.google.enterprise.adaptor;
+
+import com.google.common.base.Strings;
+
+import java.io.OutputStream;
+import java.lang.reflect.Constructor;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * The content transform factory holds all document content transformers
+ * and puts them in series connection.
+ *
+ * @author Dominik Weidenfeld (dominik.weidenfeld@twt.de)
+ */
+public class ContentTransformFactory {
+
+  private static final Logger LOG =
+      Logger.getLogger(ContentTransformFactory.class.getName());
+
+  private List<Map<String, String>> contentTransformers;
```

Comment:
Let's store List<Constructor<DocumentContentTransform>>

Let's have the ContentTransformFactory constructor go thru the config and call Class.forName().

This way we will catch many incorrect configratiuons at creation of this class' instance rather then when they are attempted to be used inside createPipeline call.

This also has the pleasent minor affect that we no longer would be keeping a reference to passed in config parameter which could be changed by caller (it isn't but reader would no longer have to know that).